### PR TITLE
added policy_iteration to mdp NOTEBOOK

### PR DESCRIPTION
**Pull request by sofmonk**
added policy_iteration to mdp notebook

### CODE REVIEW

**Comment by Chipe1** [comment]
I'm not sure what this adds to the notebook. I could see some '\n' removed, 'Widget Javascript not detected' errors and code from the mdp.py
Not sure why there are JS errors. We must use %psource to present code from the repo because any change in the code would need to be explicitly copied to the notebook otherwise.
Does anyoune know of a better method to view changes in a notebook. Looking at the diff on github is quite confusing.

**Comment by sofmonk** [comment]
@Chipe1 You may click on the **view** button on the _files changed page_, though it doesn't highlight changes, but you can compare with previous notebook to see what was added.

File: mdp.ipynb [hunk 1]
```diff
@@ -467,6 +483,102 @@
...
+    "The `policy_iteration(mdp)` function follows closely from fig. 17.7 pseudocode.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def policy_iteration(mdp):\n",
```
Comment (Chipe1) [inline mdp.ipynb]: Use %psource instead of pasting the code from mdp.py

File: mdp.ipynb [hunk 2]
```diff
@@ -467,6 +483,102 @@
...
+    "Given a policy `pi` , the `policy_evaluation` function calculates the utility of each state if `pi` were to be executed. The i<sup>th</sup> iteration specifies the action to be taken in state `s`. We can perform any no. of iterations (`k=` 50 here) to give a reasonably good approximation of utilities."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def policy_evaluation(pi, U, mdp, k=50):\n",
```
Comment (Chipe1) [inline mdp.ipynb]: Use %psource instead of pasting the code from mdp.py

File: mdp.ipynb [hunk 3]
```diff
@@ -435,14 +423,42 @@
    },
    "outputs": [
     {
+     "name": "stderr",
```
Comment (Chipe1) [inline mdp.ipynb]: Can these errors be avoided?
I get the same error but only once. Maybe removing it through text editor could work as a temporary fix

File: mdp.ipynb (hunk 2 above)
Comment (norvig) [inline mdp.ipynb]: I agree -- we need to use %psource so that we don't have the same code repeated in two places.

Comment (norvig) [inline mdp.ipynb]: I'm closing this for now.

**Pull request closed without merging**